src/Components/AlterOperation.php

```diff
@@ -279,16 +283,35 @@ public static function parse(Parser $parser, TokensList $list, array $options =
                     }
                 } elseif (! self::checkIfTokenQuotedSymbol($token)) {
                     if (! empty(Parser::$STATEMENT_PARSERS[$token->value])) {
-                        // We have reached the end of ALTER operation and suddenly found
-                        // a start to new statement, but have not find a delimiter between them
+                        // We want to get the next non-comment and non-space token after $token
+                        // therefore, the first getNext call will start with the current $idx which's $token,
+                        // will return it and increase $idx by 1, which's not guaranteed to be non-comment
+                        // and non-space, that's why we're calling getNext again.
+                        // In order to get back to the original value of idx, we kept it in $currentID
+
+                        $currentTokenID = $list->idx;
+                        $list->getNext();
+                        $nextToken = $list->getNext();
 
-                        if (! ($token->value === 'SET' && $list->tokens[$list->idx - 1]->value === 'CHARACTER')) {
+                        if ($token->value === 'CHARACTER SET'){
+                        // Reverting the changes we made in the beginning
+                        $list->idx = $currentTokenID;
```

Comment:
```suggestion
                            // Reverting the changes we made in the beginning
                            $list->idx = $currentTokenID;
```
Some spacing is missing :)

Reply:
Gonna commit and squash after finishing your review

Reply:
Thanks, back to working on this PR, will test this in minutes and do fake merges to see how it looks

Reply:
Awesome, I've squashed the last two commits, kept the first commit not squashed for better commits history, since they're kinda fixing different issues, if you want me to squash all the commits, let me know.

Reply:
All looks good ! 👏🏻